modules/cnf-performing-platform-verification-latency-tests.adoc

```diff
@@ -0,0 +1,756 @@
+
+Ran 1 of 194 Specs in 365.797 seconds
+FAIL! -- 0 Passed | 1 Failed | 0 Pending | 193 Skipped
+--- FAIL: TestTest (366.08s)
+FAIL
+----
+<1> You can configure the latency threshold by using the `MAXIMUM_LATENCY` or the `HWLATDETECT_MAXIMUM_LATENCY` environment variables.
+<2> The maximum latency value measured during the test.
+
+[id="cnf-performing-end-to-end-tests-capturing-results-hwlatdetect"]
+==== Capturing the results
+
+You can capture the following types of results:
+
+* Rough results that are gathered after each run to create a history of impact on any changes made throughout the test.
+
+* The combined set of the rough tests with the best results and configuration settings.
+
+.Example of good results
+[source, terminal]
+----
+hwlatdetect: test duration 3600 seconds
+detector: tracer
+parameters:
+Latency threshold: 10us
+Sample window: 1000000us
+Sample width: 950000us
+Non-sampling period: 50000us
+Output File: None
+
+Starting test
+test finished
+Max Latency: Below threshold
+Samples recorded: 0
+----
+
+The `hwlatdetect` tool only provides output if the sample exceeds the specified threshold.
+
+.Example of bad results
+[source, terminal]
+----
+hwlatdetect: test duration 3600 seconds
+detector: tracer
+parameters:Latency threshold: 10usSample window: 1000000us
+Sample width: 950000usNon-sampling period: 50000usOutput File: None
+
+Starting tests:1610542421.275784439, inner:78, outer:81
+ts: 1610542444.330561619, inner:27, outer:28
+ts: 1610542445.332549975, inner:39, outer:38
+ts: 1610542541.568546097, inner:47, outer:32
+ts: 1610542590.681548531, inner:13, outer:17
+ts: 1610543033.818801482, inner:29, outer:30
+ts: 1610543080.938801990, inner:90, outer:76
+ts: 1610543129.065549639, inner:28, outer:39
+ts: 1610543474.859552115, inner:28, outer:35
+ts: 1610543523.973856571, inner:52, outer:49
+ts: 1610543572.089799738, inner:27, outer:30
+ts: 1610543573.091550771, inner:34, outer:28
+ts: 1610543574.093555202, inner:116, outer:63
+----
+
+The output of `hwlatdetect` shows that multiple samples exceed the threshold. However, the same output can indicate different results based on the following factors:
+
+* The duration of the test
+* The number of CPU cores
+* The BIOS settings
+
+[WARNING]
+====
+Before proceeding with the next latency test, ensure that the number measured by `hwlatdetect` meets the required threshold. Fixing latencies introduced by hardware might require you to contact the system vendor support.
+====
+
+[id="cnf-performing-end-to-end-tests-running-cyclictest"]
+=== Running cyclictest
+
+The `cyclictest` tool measures the real-time kernel scheduler latency on the specified CPUs.
+
+.Prerequisites
+
+* You have logged into `registry.redhat.io` with your Customer Portal credentials.
+* You have installed the real-time kernel in the cluster.
+* You have applied a cluster performance profile by using Node Tuning Operator.
```

Comment:
same note regaing versions, this is relevant only for 4.11 and above. for <4.11 we use Performance addon operator